lib/controllers/ephemeral/local_auth_controller.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:get/get.dart';
+import 'package:local_auth/local_auth.dart';
+import 'package:pispapp/controllers/ephemeral/pin_entry_controller.dart';
+
+class LocalAuthController extends GetxController {
+
+  // Timer for grace period
+  static const Duration gracePeriod = Duration(seconds: 10);
+  DateTime _timestamp = DateTime.now().subtract(gracePeriod);
+
+  // Used to prevent duplicates when resuming onto a verification screen
+  bool verificationInProgress = false;
+
+  Future<void> _onUserVerificationNeeded() async {
+    // If already happening, do not open more verification screens
+    if(verificationInProgress) {
+      return;
+    }
+
+    verificationInProgress = true;
+    final localAuth = LocalAuthentication();
+    try {
+      final bool didAuth = await localAuth.authenticateWithBiometrics(
+          localizedReason: 'Please verify your identity',
+          stickyAuth: true,
+          useErrorDialogs: false);
+
+      // Use PIN screen if biometric did not validate
+      // i.e. User hit cancel
+      if(!didAuth) {
+        Get.find<PINEntryController>().onFallbackVerificationNeeded();
+      }
+      else {
+        verificationInProgress = false; // Completed verification
+      }
+    } catch(e) {
+      Get.find<PINEntryController>().onFallbackVerificationNeeded();
+    }
+  }
+
+  void appWasPaused() =>_timestamp = DateTime.now();
+  void appWasResumed() {
```

Comment:
This function can only be called after calling `appWasPaused` right? Perhaps can throw an error in case it is called without the app being paused instead of using a default value?

Reply:
What do you mean default value? Do you mean check with some kind of sentinel `wasPaused` or something?

Reply:
Yup, currently you have a default value for the `_timestamp` which is set during the class initialization right?

Reply:
Oh sorry had some misunderstanding of what you are trying to do here. Besides, I still think using a default value is not really a good idea here (since its supposed to be a non-existent time)... would it be better to use null or optional value to differentiate the case where it has a previous timestamp and the case where it doesn't? any thoughts?

Reply:
Yeah I think null is fine. The initial value is kind of hacky anyways. I'll make it null.